root/layout/components/Search.js

```diff
@@ -9,38 +9,59 @@
 
 import React from 'react';
 
+import Frag from '../../components/Frag';
 import * as manifest from '../../static/manifest';
 import * as DBDefs from '../../static/scripts/common/DBDefs';
-import {l, lp} from '../../static/scripts/common/i18n';
+import {compare, l, lp} from '../../static/scripts/common/i18n';
 
-const TYPE_OPTIONS = {
-  artist:         l('Artist'),
-  release_group:  l('Release Group'),
-  release:        l('Release'),
-  recording:      l('Recording'),
-  work:           l('Work'),
-  label:          l('Label'),
-  area:           l('Area'),
-  place:          l('Place'),
-  annotation:     l('Annotation'),
-  cdstub:         l('CD Stub'),
-  editor:         l('Editor'),
-  tag:            lp('Tag', 'noun'),
-  instrument:     l('Instrument'),
-  series:         lp('Series', 'singular'),
-  event:          l('Event'),
-  doc:            DBDefs.GOOGLE_CUSTOM_SEARCH ? l('Documentation') : null,
-};
+const TYPE_OPTION_GROUPS = [
+  {
+    artist:        l('Artist'),
```

Comment:
Oops, I noticed one more issue which isn't strictly related to MBS-9672, but maybe you'd like to fix it as part of this PR. :) https://tickets.metabrainz.org/browse/MBS-9673

We have `N_l` and `N_lp` functions in i18n.js that you can import, but they'll be slightly annoying to use due to series...you'd need `{key === 'series' ? lp(text) : l(text)}` in `searchOptions` below since the output of `N_l` has to be fed into `l`, and likewise for `N_lp` -> `lp`.

Another alternative is just turning all of the strings in `TYPE_OPTION_GROUPS` into functions like `artist: () => l('Artist')` so we can delay their evaluation.

Reply:
A cool idea might be to have the `N_*` functions return objects with a `.translate()` method that calls the appropriate `l*` function. Sometime later maybe.

Reply:
I still don’t get the exact causes of [MBS-9673](https://tickets.metabrainz.org/browse/MBS-9673) as I cannot reproduce it with my local setup, although I guess it is some kind of caching issue.
1. Why German translations are currently sometimes returned (see [comment](https://tickets.metabrainz.org/browse/MBS-9673?focusedCommentId=45971#comment-45971))?
2. Why entity tabs are not affected by this bug whereas they use `l` instead of `N_l` too?

Apart from this bug, we should probably be using `i18n.strings.entityName` for core entity type search options. That would require to change `i18n.strings.*` implementation too.

Reply:
So basically, since `TYPE_OPTION_GROUPS` is defined at the top level, all of these `l()` calls execute *when the module loads*. In Node.js, once a module is `require()`d, it's executed once and cached - if you `require()` it again (anywhere, at any point in the same process), it just returns the cached exports from the first require.

So if the entity type options are sometimes in another language, it's because the *very first request* which `require()`d that module was from a user with that language enabled.

Note that if you have `DEVELOPMENT_SERVER` set to `1`, this won't be reproducible, because we clear `require.cache` (which holds the cached modules) after each request. So on every request, it will execute every module again. `git grep clearRequireCache` will show you how that works.

Not sure which part of the entity tabs you're referring to - in root/components/EntityTabs.js I see `tabLinkNames` using `N_l`.

Reply:
Steps to reproduce the different language being displayed:

1. Disable `DEVELOPMENT_SERVER`.
2. Switch to a language other than English.
3. Restart root/server.js.
4. Reload the page. The entity types will now be in the language you selected.
5. Switch the language back to English. Note the dropdown is still in the previous language, because the module is still cached from the first request.
6. Restart root/server.js again.
7. Reload the page. The types are in English now, as expected.

Reply:
Thanks! I was referring to _Overview_ and all, but they are translated from within the `buildLinks` function.